op-service/dial/static_rollup_provider.go

```diff
@@ -10,7 +10,9 @@ import (
 // RollupProvider is an interface for providing a RollupClient
 // It manages the lifecycle of the RollupClient for callers
 type RollupProvider interface {
-	// RollupClient(ctx) returns the underlying sources.RollupClient pointing to the L2 rollup consensus node
+	// RollupClient(ctx) returns the underlying sources.RollupClient pointing to the L2 rollup consensus node.
+	// Note: ctx should be a lifecycle context as client selection may involve multiple network operations,
```

Comment:
Please add the same note to the `EthClient` function of the `L2EndpointProvider` interface.